plutus-playground/plutus-playground-server/usecases/Vesting.hs

```diff
@@ -53,21 +52,23 @@ vestFunds vst value = do
         vd =  VestingData (validatorScriptHash vst) 0
     void (payToScript contractAddress value dataScript)
 
--- | Retrieve some of the vested funds.
-retrieveFunds :: 
-    Vesting
-    -> VestingData -- ^ Value that has already been taken out
-    -> TxOutRef'  -- ^ Transaction output locked by the vesting validator script
-    -> Ledger.Value -- ^ Value we want to take out now
-    -> MockWallet ()
-retrieveFunds vs vd r vnow = do
-    oo <- ownPubKeyTxOut vnow
-    let val = validatorScript vs
-        o   = scriptTxOut remaining val (DataScript $ Ledger.lifted vd')
-        remaining = totalAmount vs - vnow
-        vd' = vd {vestingDataPaidOut = vnow + vestingDataPaidOut vd }
-        inp = scriptTxIn r val Ledger.unitRedeemer
-    void (signAndSubmit (Set.singleton inp) [oo, o])
+-- | Register this wallet as the owner of the vesting scheme. At each of the
+--   two dates (tranche 1, tranche 2) we take out the funds that have been 
+--   released so far.
+--   This function has to be called before the funds are vested, so that the 
+--   wallet can start watching the contract address for changes.
+registerVestingOwner :: Vesting -> MockWallet ()
+registerVestingOwner v = do
+    ourPubKey <- ownPubKey
+    let 
+        o = vestingOwner v
+        addr = Ledger.scriptAddress (validatorScript v)
+    _ <- if o != ourPubKey 
+         then otherError "Vesting scheme is not owned by this wallet" 
+         else startWatching addr
+    
+    -- register (tranche2Trigger v) (tranche2Handler v)
```

Comment:
commented out code

Reply:
Although I would expect to see a handler for tranche 1?

Reply:
the problem is that to take out tranche 1 you need to put the rest of the funds in a pay-to-script output with the new data script, and currently that's only possible if work with the `Map`s and `Set`s that make up the txn directly. I'm thinking about better support for that in the `WalletAPI` module but I'm not sure yet how to do it.

The contract still works with the actions that are there currently, and if you add the right imports etc. you can simulate taking out tranche 1 in the Playground, but in my opinion it adds too much clutter to the sample contract.

Reply:
Well, can we have some explanation in the code? Otherwise this is a bit baffling for the reader.

Reply:
I was hoping to add that before anyone would look at the code next week ;) but yeah I'll write a note